pkg/awsclient/iam.go

```diff
@@ -186,9 +189,7 @@ func ListIAMRoles(reqLogger logr.Logger, client Client) ([]*iam.Role, error) {
 			return nil, err
 		}
 
-		for _, role := range output.Roles {
-			iamRoleList = append(iamRoleList, role)
-		}
+		iamRoleList = append(iamRoleList, output.Roles...)
```

Comment:
That's an interesting replacement

Reply:
Yeah!  I learned some things running through all the fixes. It's been great.